tools/modify_master.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/python
+
+import shakedown
+import logging
+
+
+# Methods to modify the envvar settings of a mesos master and restart
+# the master process after they are modified
+
+logger = logging.getLogger(__name__)
+
+
+def set_master_envvar(envvar, value):
+    as_dict = {envvar: value}
+    modify_envvars(lambda envvars: envvars.update(as_dict))
+
+
+def set_master_envvars(new_envvars_as_dict):
+    modify_envvars(lambda envvars: envvars.update(new_envvars_as_dict))
+
+
+def remove_master_envvar(envvar):
+    modify_envvars(lambda envvars: envvars.pop(envvar, None))
+
+
+def modify_envvars(modifier):
+    logger.info("Modifying envvars on master node...")
+    # Get the current set of master envvars
+    success, out = shakedown.run_command_on_master('cat /opt/mesosphere/etc/mesos-master')
+    if success is not True:
+        logger.info('Unable to get current envvars from master: {}'.format(out))
+        raise RuntimeError("Unable to get current envvars")
+
+    logger.info("Current envvars:\n{}".format(out))
+    envvars, commented = process_envvars(out)
+    modifier(envvars)
+    write_envvars(envvars, commented)
+    restart_master()
+    logger.info("Modification complete.")
+
+
+def process_envvars(input):
+    logger.info("Processing envvars...")
+    lines = input.split('\n')
+
+    envvars = {}
+    commented = []
+    for line in lines:
+        if '#' in line:
+            commented.append(line)
+            continue
+        if '=' not in line:
+            continue
+        bits = line.split('=')
+        if len(bits) != 2:
+            raise ValueError('Envvar file had badly formatted line: {}\nfile: {}'.format(
+                line, input))
```

Comment:
The slightly ugly 
`var, value = line.slit('=', 1)`
handles this.  (Ugly, becauase wtf is that 1?  The split count.  maxsplit=1 is a syntax error because the C implementation is lazy.)
I'd bet a value with an equals in it is technically valid, so you can probably drop the error handling.